BLDC.py

```diff
@@ -17,21 +19,24 @@ def __init__(self, pwmPin, directionPin, frequency):
             self.pwmCtrl = PWM.Signal(pwmPin, frequency)
 
     def setDirection(self, direction):
+        self.stop()
+        time.sleep(self.delay)
         if direction == 'CW':
             GPIO.write(self.directionPin, 'LOW')
         elif direction == 'CCW':
             GPIO.write(self.directionPin, 'HIGH')
         else:
             raise ValueError("Could not set motor direction, expected either: 'CW' or 'CCW'")
+        time.sleep(self.delay)
 
     def setSpeed(self, speed):                     #speed in percentage
-        if speed < 50:
-            raise ValueError("Speed is too low, please set it to a value higher than 50%")
-        else:
-            self.speed = speed
+        self.stop()
+        self.speed = speed
+        time.sleep(self.delay)
 
     def start(self):
         self.pwmCtrl.begin(self.speed)
 
     def stop(self):
+        self.speed = 0
```

Comment:
do we need to sleep every time we call `stop()`? then we can just add it here.

Reply:
Yes, I can do that to simply things, but there still needs to be a delay at the end of setDirection() because of the pin change in the direction pin